addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -0,0 +1,581 @@
+            if (thing.getThingTypeUID().equals(THING_TYPE_GATEWAY)) {
+                String id = thing.getConfiguration().get("id").toString();
+                updateGatewayState(thing, id);
+            } else {
+                String url = thing.getConfiguration().get("url").toString();
+                updateThingState(thing, url);
+            }
+        }
+    }
+
+    public void updateThingState(Thing thing, String url) {
+        SomfyTahomaThingHandler handler = (SomfyTahomaThingHandler) thing.getHandler();
+        if (handler.getStateName() != null) {
+            State state = getState(handler, url);
+            if (state == null || state.equals(UnDefType.UNDEF)) {
+                return;
+            }
+
+            if (state.equals(UnDefType.NULL)) {
+                //relogin
+                login();
+                state = getState(handler, url);
+            }
+
+            for (Channel channel : thing.getChannels()) {
+                updateState(channel.getUID(), state);
+            }
+        }
+
+    }
+
+    public void updateChannelState(SomfyTahomaThingHandler handler, ChannelUID channelUID, String url) {
+        if (handler.getStateName() != null) {
+            State state = getState(handler, url);
+            if (state == null || state.equals(UnDefType.UNDEF)) {
+                return;
+            }
+
+            if (state.equals(UnDefType.NULL)) {
+                //relogin
+                login();
+                state = getState(handler, url);
+            }
+
+            updateState(channelUID, state);
+        }
+    }
+
+    private void logout() {
+        try {
+            sendToTahomaWithCookie(TAHOMA_URL + "logout");
+            cookie = "";
+        } catch (Exception e) {
+            logger.error("Cannot send logout command!", e);
+        }
+    }
+
+    public String readResponse(InputStream response) throws Exception {
+        String line;
+        StringBuilder body = new StringBuilder();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(response));
+
+        while ((line = reader.readLine()) != null) {
+            body.append(line).append("\n");
+        }
+        line = body.toString();
+        logger.trace("Response: {}", line);
+        return line;
+    }
+
+    public InputStream sendToTahomaWithCookie(String url) throws Exception {
+        logger.trace("Sending GET to Tahoma url: {}", url);
+        URL cookieUrl = new URL(url);
+        HttpsURLConnection connection = (HttpsURLConnection) cookieUrl.openConnection();
+        connection.setDoOutput(false);
+        connection.setRequestMethod("GET");
+        setConnectionDefaults(connection);
+        connection.setRequestProperty("Cookie", cookie);
+
+        return connection.getInputStream();
+    }
+
+    public InputStream sendDataToTahomaWithCookie(String url, byte[] postData) throws Exception {
+        logger.trace("Sending POST to Tahoma to url: {} with data: {}", url, postData);
```

Comment:
Could you change the sendDataToTahomaWithCookie() method to accept (String url, String postData) and do the urlParameters.getBytes(StandardCharsets.UTF_8); call inside that method?

This way, in your logger.trace(), you would see the String that is being sent to the server instead of an array of int values 😄